File: lab9/2.js
class MyStack {
  constructor() {
    this.a = [];
  }

  push(x) {
    this.a.push(x);
    for (let i = 0; i < this.a.length - 1; i++) {
      this.a.push(this.a.shift());
    }
  }

  pop() {
    return this.a.shift();
  }

  top() {
    return this.a[0];
  }

  empty() {
    return this.a.length === 0;
  }
}

File: lab9/2.cpp
class MyStack {
public:
    queue<int> a;

    MyStack() {
        
    }
    
    void push(int x) {
        a.push(x);
        int n = a.size(); 

        for(int i=0; i<n-1; i++){
            a.push(a.front());
            a.pop();
        }
    }
    
    int pop() {
        int result = a.front();
        a.pop();

        return result;
    }
    
    int top() {
        return a.front();
    }
    
    bool empty() {
        return a.empty();
    }
};